### ISSUE TITLE
stuck when trying to download a file for the first time

### ISSUE THREAD

**Issue by caradave** [polito/nuova-pagina-studenti#19]
**Describe the bug**
The first time you download a file from the "materiale" section, it does not bring you back to the "materiale" page of the course. Instead, it stays on a loading page as if it’s trying to log in to the site.

**To Reproduce**
1. Delete the cookies of the site.
2. Download a file from the "materiale" course.
**Expected behavior**
It should bring you back to the "materiale" page.

Screenshots
![image](https://github.com/user-attachments/assets/0dee2030-e962-49ad-8e60-8863e2e1ef0e)

**Desktop (please complete the following information):**
OS: Windows
Browser: Chrome
Version: 131.0.6778.205

**Additional context**
If you manually go back to the page, then it works fine.

**Comment by garzena** [comment]
The file was being downloaded in the background. However, an intermediate page was added to the download flow for the first request.

**Comment by caradave** [comment]
that's true, and now i'm no loger stuck on the login page, but it does not bring me back to the materiale page, it brings me to this page 
![image](https://github.com/user-attachments/assets/fd601f5d-d123-41b4-8137-43ed86b0a3e2)
